model/task/task.go

```diff
@@ -499,14 +503,19 @@ func (t *Task) IsDispatchable() bool {
 // IsHostDispatchable returns true if the task should run on a host and can be
 // dispatched.
 func (t *Task) IsHostDispatchable() bool {
-	return t.IsHostTask() && t.WillRun()
+	return t.IsHostTask() && t.WillRun() && !t.IsStuckTask()
```

Comment:
See my comment about `assignNextAvailableTask`, but I don't think this would ever actually run. If the host has already been assigned a task, it'll never hit this case.